### ISSUE TITLE
0.4.10 cleanup behavior issue

### ISSUE THREAD

**Issue by leshik** [dokku/dokku#1852]
I had some stopped containers before upgrading to 0.4.10 and found that they were wiped out after upgrade. For example, one was with redis, now `dokku redis:list` shows errors:

Error: No such image or container: dokku.redis.smod
Error: No such image or container: c1b76e0280c4f82ac4b7e0e6875af549a153f9d9c197141a2ea7d0dc6e304e0a
Error: No such image or container: c1b76e0280c4f82ac4b7e0e6875af549a153f9d9c197141a2ea7d0dc6e304e0a
Error: No such image or container: c1b76e0280c4f82ac4b7e0e6875af549a153f9d9c197141a2ea7d0dc6e304e0a
Error: No such image or container: c1b76e0280c4f82ac4b7e0e6875af549a153f9d9c197141a2ea7d0dc6e304e0a
Error: No such image or container: c1b76e0280c4f82ac4b7e0e6875af549a153f9d9c197141a2ea7d0dc6e304e0a
NAME  VERSION      STATUS   EXPOSED PORTS  LINKS
smod  stopped      -        smod

The data is still present in /var/lib/dokku/services/redis/, but the container can't be started:

`dokku redis:start smod`
-----> Starting container
       No container exists for smod

What is the proper way to restore the service?

**Comment by josegonzalez** [comment]
Please re-file this issue in the [dokku-redis](https://github.com/dokku/dokku-redis/issues) plugin issue tracker.

Note that this seems similar to the issue in https://github.com/dokku/dokku-mariadb/pull/32.

**Comment by leshik** [comment]
@josegonzalez I'm seeing this with postgres plugin also. I didn't try others, but I'm pretty sure that other plugins are affected. I think it's more general issue than just redis plugin.

**Comment by josegonzalez** [comment]
Right but its not an issue with the Dokku core project itself.

**Comment by jvanbaarsen** [comment]
@josegonzalez Has there been a big update to 0.4.10? Since maybe some plugin hook or something has changed, and that is causing this issue?

**Comment by josegonzalez** [comment]
No. This issue predates any specific version of dokku, and is caused by users manually stopping those containers. `dokku cleanup` has *always* removed stopped containers, and since that is run before a deploy, the containers were removed, even though they weren't application containers.

The plugins *should* handle starting a new container up properly - if the id doesn't exist, just start a new one, as all state is (hopefully) persisted to disk - but they don't. There is a PR opened to fix this in mariadb, but it's not yet passing tests. Once it does, we can port it to every other repo.

Note that in 0.4.10, we've started labeling app containers managed by dokku. In some future minor release (maybe 0.5.0?) we will only cleanup containers that have this specific label, and leave all other containers stopped.
